### ISSUE TITLE
undesirable_function_linter() should look for function calls only

### ISSUE THREAD

**Issue by jennybc** [jimhester/lintr#556]
I think the `undesirable_function_linter()` should only look for function calls. The context is that I want to use lintr to keep watch for base file system functions creeping into usethis (https://github.com/r-lib/usethis/issues/531, https://github.com/r-lib/fs/issues/85). I can *almost* do what I want with the `undesirable_function_linter()`.

But it reports tons of false positives, i.e. warns when it should not.

`dir()` is one of the functions I'm linting for. But throughout the package, below both `R/` and `tests/`, the name `dir` is frequently used for the path to a directory. So the linter reports dozens of false warnings.

Here's a small reprex with current lintr vs. after the change in my PR (#557):

``` r
library(lintr)

lint_this <- function(this, ...) {
  file <- tempfile()
  on.exit(unlink(file))
  writeLines(this, con = file, sep = "\n")
  lint(file, ...)
}

linter <- undesirable_function_linter(fun = c("dir" = NA))

# warning is correct: I'm calling dir()
lint_this('dir(".")', linter)
#> /private/var/folders/pv/l2mbh7j90y5_vslh_1wyqxj00000gn/T/RtmpQbfI18/file121123553fe9e:1:1: warning: Function "dir" is undesirable.
#> dir(".")
#> ^~~

# warning is WRONG: I'm defining an object named 'dir'
lint_this("dir <- 1:3", linter)
#> /private/var/folders/pv/l2mbh7j90y5_vslh_1wyqxj00000gn/T/RtmpQbfI18/file121121d71526c:1:1: warning: Function "dir" is undesirable.
#> dir <- 1:3
#> ^~~

devtools::load_all("~/rrr/lintr")
#> ℹ Loading lintr

linter <- undesirable_function_linter(fun = c("dir" = NA))

# warning is correct: I'm calling dir()
lint_this('dir(".")', linter)
#> /private/var/folders/pv/l2mbh7j90y5_vslh_1wyqxj00000gn/T/RtmpQbfI18/file121127ada3f8d:1:1: warning: Function "dir" is undesirable.
#> dir(".")
#> ^~~

# lack of warning is correct: I'm defining an object named 'dir'
lint_this("dir <- 1:3", linter)
```

<sup>Created on 2020-11-01 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>
